pkg/ebpf/filters.go

```diff
@@ -38,6 +41,327 @@ type Filter struct {
 	ProcessTreeFilter *filters.ProcessTreeFilter
 	Follow            bool
 	NetFilter         *NetIfaces
+
+	CheckEventsInUserSpace bool
+	CheckSetsInUserSpace   bool
+}
+
+func NewFilterScope() FilterScope {
+	return FilterScope{
+		ID:                     0,
+		EventsToTrace:          []events.ID{},
+		SetsToTrace:            map[string]bool{},
+		UIDFilter:              filters.NewBPFUInt32Filter(UIDFilterMap),
+		PIDFilter:              filters.NewBPFUInt32Filter(PIDFilterMap),
+		NewPidFilter:           filters.NewBoolFilter(),
+		MntNSFilter:            filters.NewBPFUIntFilter(MntNSFilterMap),
+		PidNSFilter:            filters.NewBPFUIntFilter(PidNSFilterMap),
+		UTSFilter:              filters.NewBPFStringFilter(UTSFilterMap),
+		CommFilter:             filters.NewBPFStringFilter(CommFilterMap),
+		ContFilter:             filters.NewBoolFilter(),
+		NewContFilter:          filters.NewBoolFilter(),
+		ContIDFilter:           filters.NewContainerFilter(CgroupIdFilterMap),
+		RetFilter:              filters.NewRetFilter(),
+		ArgFilter:              filters.NewArgFilter(),
+		ProcessTreeFilter:      filters.NewProcessTreeFilter(ProcessTreeFilterMap),
+		Follow:                 false,
+		NetFilter:              &NetIfaces{Ifaces: []string{}},
+		CheckEventsInUserSpace: false,
+		CheckSetsInUserSpace:   false,
+	}
+}
+
+const MaxFilterScopes = 64
+
+func isIDInRange(id int) bool {
+	return id >= 0 && id < MaxFilterScopes
+}
+
+type FilterScopes struct {
+	filterScopesArray        [MaxFilterScopes]*FilterScope
+	filterScopesMap          map[*FilterScope]int
+	filterScopesUserSpaceMap map[*FilterScope]int
+
+	uidFilterMin             uint64
+	uidFilterMax             uint64
+	pidFilterMin             uint64
+	pidFilterMax             uint64
+	uidFilterableInUserSpace bool
+	pidFilterableInUserSpace bool
+}
```

Comment:
Insert a comment regarding the future need for locking as scopes will change at runtime.

Reply:
Add fields with pre-calculated values to reduce processing in hot paths, i.e.: `ContainerFilterEnabled()`, `IsFilterNetSet()`.